src/components/editor/Bindings/FieldSelection/index.tsx

```diff
@@ -343,15 +325,70 @@ function FieldSelectionViewer({ collectionName }: Props) {
         setSelectionSaving,
     ]);
 
+    useMount(() => {
+        const existingSettings = tableSettings ?? {};
+
+        if (
+            !tableSettings ||
+            !Object.hasOwn(tableSettings, TablePrefixes.fieldSelection)
+        ) {
+            setTableSettings({
+                ...existingSettings,
+                [TablePrefixes.fieldSelection]: {
+                    hiddenColumns: optionalColumns.map((column) =>
+                        column.headerIntlKey
+                            ? intl.formatMessage({ id: column.headerIntlKey })
```

Comment:
We should not be translating this. It should just use the key itself as the key since it is semi-guaranteed to be unique and a bit safer.

Reply:
A TODO must have been dropped while refactoring the code as this was intended to be temporary. Will update.